### ISSUE TITLE
I would like to be able to ignore tag-only commits

### ISSUE THREAD

**Issue by jchesterpivotal** [concourse/git-resource#91]
I can use `ignore_paths` to ignore changing files.

But it's less clear how to use `tag_filter` to ignore tag-only commits. I don't want to _include_ particularly-tagged commits. I want to _exclude_ some or all tagged commits.

**Comment by vito** [comment]
Not sure what you mean by tag-only commits. No-op commits made with `--allow-empty` that have a tag pointing to them?

**Comment by jchesterpivotal** [comment]
We watch a lot of github projects with varying levels of tagging discipline. Some will only tag releases, which is nice. Some will tag releases, pre-releases, the name of their kids, the time of day, nonsensical versions worthy of Dr Seuss and so on and so forth.

For the latter kind, we want to be able to exclude tags that don't fit a release pattern. I wish I'd kept the motivating example at hand, or linked to it, because I don't remember at the moment :|

**Comment by d** [comment]
The checking behavior is to list new commits. Tags, if I understand them correctly, are not commits.

**Comment by jchesterpivotal** [comment]
Tags either *point* to commits (lightweight tags) or they are *commit-like* (annotated tags).

Either way, we'd like to be able to filter by pattern.

**Comment by d** [comment]
Did this work for you?
```
source:
  tag_filter: 'v*'
```

**Comment by jchesterpivotal** [comment]
I sincerely do not recall. I do remember reading the readme pretty forensically.

**Comment by jchesterpivotal** [comment]
OK, here is the example:

https://github.com/aspnet/Home/releases

I want to include any tag that *does not include* `preview` in it.

In theory I can match on `\d\.\d\.\d` but it would be much clearer to be able to exclude tags.

**Comment by vito** [comment]
Agreed on the principle that it's more obvious to teammates to explicitly say "skip preview releases".
